source/extensions/filters/http/json_to_metadata/filter.h

```diff
@@ -0,0 +1,117 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "envoy/extensions/filters/http/json_to_metadata/v3/json_to_metadata.pb.h"
+#include "envoy/json/json_object.h"
+#include "envoy/server/filter_config.h"
+#include "envoy/stats/scope.h"
+#include "envoy/stats/stats.h"
+#include "envoy/stats/stats_macros.h"
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/extensions/filters/http/common/pass_through_filter.h"
+
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JsonToMetadata {
+
+// clang-format off
+#define ALL_JSON_TO_METADATA_FILTER_STATS(COUNTER)                \
+  COUNTER(rq_success)                                             \
+  COUNTER(rq_mismatched_content_type)                             \
+  COUNTER(rq_no_body)                                             \
+  COUNTER(rq_too_large_body)                                      \
+  COUNTER(rq_invalid_json_body)
+// clang-format on
+
+struct JsonToMetadataStats {
+  ALL_JSON_TO_METADATA_FILTER_STATS(GENERATE_COUNTER_STRUCT)
+};
+
+using ProtoRule = envoy::extensions::filters::http::json_to_metadata::v3::JsonToMetadata::Rule;
+using KeyValuePair =
+    envoy::extensions::filters::http::json_to_metadata::v3::JsonToMetadata::KeyValuePair;
+using ValueType = envoy::extensions::filters::http::json_to_metadata::v3::JsonToMetadata::ValueType;
+
+class Rule {
+public:
+  Rule(const ProtoRule& rule);
+  const ProtoRule& rule() const { return rule_; }
+  const std::vector<std::string>& keys() const { return keys_; }
+
+private:
+  const ProtoRule rule_;
+  std::vector<std::string> keys_;
+};
+
+using Rules = std::vector<Rule>;
+
+class FilterConfig {
```

Comment:
Can the classes here be documented some more?

Reply:
Do you want more doc for member functions or just classes?